fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/literal/VarcharLiteral.java

```diff
@@ -63,4 +67,28 @@ public LiteralExpr toLegacyLiteral() {
     public String toString() {
         return "'" + value + "'";
     }
+
+    @Override
+    protected Expression uncheckedCastTo(DataType targetType) throws AnalysisException {
+        if (getDataType().equals(targetType)) {
+            return this;
+        }
+        if (targetType.isDateType()) {
+            return convertToDate(targetType);
+        } else if (targetType.isIntType()) {
+            return new IntegerLiteral(Integer.parseInt(value));
+        }
+        //todo other target type cast
```

Comment:
remove this TODO, and add a comment to this function: temporary way to process type coercion in TimestampArithmetic, should be replaced by TypeCoercion rule.